test/Microsoft.AspNetCore.Razor.Evolution.Test/IntegrationTests/IntegrationTestBase.cs

```diff
@@ -109,5 +109,36 @@ protected void AssertIRMatchesBaseline(DocumentIRNode document)
             var baseline = testFile.ReadAllText().Split(new char[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
             RazorIRNodeVerifier.Verify(document, baseline);
         }
+
+        protected void AssertCSharpDocumentMatchesBaseline(RazorCSharpDocument document)
+        {
+            if (Filename == null)
+            {
+                var message = $"{nameof(AssertCSharpDocumentMatchesBaseline)} should only be called from an integration test ({nameof(Filename)} is null).";
+                throw new InvalidOperationException(message);
+            }
+
+            var baselineFilename = Path.ChangeExtension(Filename, ".codegen.cs");
```

Comment:
Was thinking we could describe that via the test class name. In this case we have a razor runtime csharp code generator test. Some duplication but very clear.